src/dmd/compiler.d

```diff
@@ -12,7 +12,59 @@
 
 module dmd.compiler;
 
+///
 struct Compiler
 {
-    const(char)* vendor; // Compiler backend name
+    /// Compiler backend name
+    const(char)* vendor;
 }
+
+/**
+ * Initializes the dmd lexer and all required components in order
+ *
+ * It is templated (so D only) to prevent
+ *  errors on local imports to things that may not exist.
```

Comment:
Not sure I understand the motivation. Exception for the private import error this compiles fine.

Reply:
As mentioned this currently fails to compile because of private access which is why using templates is a bad idea, because it will never be instantiated and tested by the compiler.
Hence, I would strongly prefer not to use templates.
If you have import problems, use `frontend.d`

Reply:
I am not seeing any compile errors. So if you are getting any, I would like to know.

This will be tested and initiated, but not in this PR. Dmd will call it, as part of its main function. I am currently using it locally and is hooked up into Intellij IDEA.

It is not appropriate to put (de)initialization functions into frontend.d. It is meant for helper functions which (de)initialization is not. The current function in there that does this, exists as a crutch, nothing more.

Reply:
Hmm, sorry I can't reproduce the build failure anymore. Maybe it was just a stale build cache.